File: nodecore-cli/src/main/java/nodecore/cli/serialization/BitcoinBlockPayload.java
package nodecore.cli.serialization;

// VeriBlock NodeCore CLI
// Copyright 2017-2020 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

import com.google.gson.annotations.SerializedName;
import nodecore.api.grpc.VeriBlockMessages;
import nodecore.api.grpc.utilities.ByteStringUtility;

public class BitcoinBlockPayload {

    public BitcoinBlockPayload(final VeriBlockMessages.GetLastBitcoinBlockReply reply) {
        header = ByteStringUtility.byteStringToHex(reply.getHeader());
        height = reply.getHeight();
        hash = ByteStringUtility.byteStringToHex(reply.getHash());
    }

    @SerializedName("header")
    public String header;

    @SerializedName("height")
    public int height;

    @SerializedName("hash")
    public String hash;

    public String getHeader() {
        return header;
    }

    public int getHeight() {
        return height;
    }

    public String getHash() {
        return hash;
    }
}

File: nodecore-cli/src/main/java/nodecore/cli/serialization/BitcoinBlockPayload.kt
// VeriBlock NodeCore CLI
// Copyright 2017-2021 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.cli.serialization

import com.google.gson.annotations.SerializedName
import nodecore.api.grpc.RpcGetLastBitcoinBlockReply
import org.veriblock.sdk.extensions.toHex

class BitcoinBlockPayload(
    reply: RpcGetLastBitcoinBlockReply
) {
    @SerializedName("header")
    val header = reply.header.toHex()

    @SerializedName("height")
    val height = reply.height

    @SerializedName("hash")
    val hash = reply.hash.toHex()
}
